bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -701,28 +701,23 @@ uint32_t PreProcessor::launchReqPreProcessing(uint16_t clientId,
                                               char *reqBuf,
                                               const concordUtils::SpanContext &span_context) {
   concord::diagnostics::TimeRecorder scoped_timer(*histograms_.launchReqPreProcessing);
-  uint32_t resultLen = 0;
   // Unused for now. Replica Specific Info not currently supported in pre-execution.
-  uint32_t replicaSpecificInfoLen = 0;
   auto span = concordUtils::startChildSpanFromContext(span_context, "bft_process_preprocess_msg");
   SCOPED_MDC_CID(cid);
   LOG_DEBUG(logger(), "Pass request for a pre-execution" << KVLOG(reqSeqNum, clientId, reqSeqNum));
-  auto status = requestsHandler_.execute(clientId,
-                                         reqSeqNum,
-                                         PRE_PROCESS_FLAG,
-                                         reqLength,
-                                         reqBuf,
-                                         maxPreExecResultSize_,
-                                         (char *)getPreProcessResultBuffer(clientId),
-                                         resultLen,
-                                         replicaSpecificInfoLen,
-                                         span);
+  bftEngine::IRequestsHandler::ExecutionRequestsQueue accumulatedRequests;
+  accumulatedRequests.push_back(bftEngine::IRequestsHandler::ExecutionRequest{
+      clientId, reqSeqNum, PRE_PROCESS_FLAG, reqLength, reqBuf, std::string(maxPreExecResultSize_, 0)});
+  requestsHandler_.execute(accumulatedRequests, cid, span);
+  const IRequestsHandler::ExecutionRequest &request = accumulatedRequests.back();
+  memcpy((char *)getPreProcessResultBuffer(clientId), request.outReply.c_str(), request.outActualReplySize);
```

Comment:
This copying may turn out to be an expensive operation. Why is it necessary?

Reply:
This buffer holds the result, if we don't copy that we will lose the result and probably we use this result.
For example, I test it without copying and the preprocessor tests have failed.

Reply:
Before your changes, why didn’t we need to copy?

PS: bare in mind those buffers can hold up to 34MB of data

Reply:
Maybe we should use a raw buffer for the result too?

Reply:
we sent this buffer to the execute before my changes so we had 1 copy, since we use string to get the result so I must have another copy.

Reply:
The result should be a raw buffer, too.

Reply:
https://github.com/vmware/concord-bft/pull/1097#discussion_r545029692
That will solve the copy.